### PR TITLE
Setting Default Profile

### PR DESCRIPTION
**Pull request by sarinali**
# Description

Verifies that the profile shows as default when it is set as the default profile.

## Bugzilla bug ID

**ID:** 1902744 
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1902744

## Type of change

- [x] New Test
- [x] New POM
- [x] Other Changes: Modified `page_base.py` to accept an additional parameter `parent_element` which would ensure that 

# How does this resolve / make progress on that bug?

Done.

# Screenshots / Explanations

N/A

# Comments / Concerns

Same 6 fail from Nightly change but these were fixed in another PR.

### CODE REVIEW

File: tests/profile/test_set_default_profile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    # verify that some profile is the default
+    profile_header_in_use = about_profiles.get_element(
+        "profile-container-item-default-header"
+    )
+    assert profile_header_in_use is not None
+
+    cur_default = -1
+
+    # find index that is the current default
+    for i in range(len(profiles)):
```
Comment (ben-c-at-moz) [inline tests/profile/test_set_default_profile.py:29]: Iterating is fine, but do we need to do it? All we need to know is that there _is_ a default profile, right? Would it be quicker to just find that element anywhere in the `about:profiles` page?

File: tests/profile/test_set_default_profile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            cur_default = i
+            break
+
+    # no default profile could be found
+    if cur_default == -1:
+        logging.warn("Could not find a currently active default profile.")
+        assert False
+
+    # select a non default profile randomly
+    profile_index = random.randint(0, len(profiles) - 1)
```
Comment (ben-c-at-moz) [inline tests/profile/test_set_default_profile.py:53]: could make a set of all the profiles that are not the current, then choose a random from that, would avoid iteration.

**Review by ben-c-at-moz — APPROVED** [review]
A couple of dorky notes from me, but the code works and looks fine.

File: tests/profile/test_set_default_profile.py (hunk 1 above)
Comment (sarinali) [reply tests/profile/test_set_default_profile.py:29]: iteration is required to find the default profile!

File: tests/profile/test_set_default_profile.py (hunk 2 above)
Comment (sarinali) [reply tests/profile/test_set_default_profile.py]: Got it. Fix was just pushed

**Review by ben-c-at-moz — APPROVED** [review]
Nice work!

**Pull request merged**